### PR TITLE
BUG Merge not behaving correctly when having `MultiIndex` with a single level

### LINKED ISSUE #52331: BUG: Unexpected Behaviour `pd.merge` using `left_on` and `right_index=True` with single level MultiIndex
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
# merge fails to join the DataFrames with a single level MultiIndex
# expected successful join on first row =='A'
import pandas as pd
left_col = ['A', 'B']
right_index = [('A',)]
left_df = pd.DataFrame({"col":left_col})
right_df = pd.DataFrame(data={"b":[100]}, index=pd.MultiIndex.from_tuples(right_index, names=["col"]))
pd.merge(left_df, right_df, left_on=["col"], right_index=True, how="left", validate="m:1")

# merge succeeds to join the DataFrames if the MultiIndex has 2 levels
import pandas as pd
left_col = ['A', 'B']
right_index = [('A','A',)]
left_df = pd.DataFrame({"col1":left_col, "col2":left_col})
right_df = pd.DataFrame(data={"b":[100]}, index=pd.MultiIndex.from_tuples(right_index, names=["col1", "col2"]))
pd.merge(left_df, right_df, left_on=["col1", "col2"], right_index=True, how="left", validate="m:1")

# merge succeeds if it's an Index
import pandas as pd
left_col = ['A', 'B']
right_index = ['A']
left_df = pd.DataFrame({"col":left_col})
right_df = pd.DataFrame(data={"b":[100]}, index=pd.Index(right_index, name="col"))
pd.merge(left_df, right_df, left_on="col", right_index=True, how="left", validate="m:1")
```

### Issue Description

In the first example, even though a list of 1 column is provided for `left_on`, matching the length of the index levels, merge fails to join successfully.
I think that given that it works for 2 levels, and that it works when there is a single level index, it should work for 1 as well, especially because it fails silently.
Alternatively, it should raise an Error if something is not correct.

### Expected Behavior

First example returns:

|   | col | b    |
|---|-----|------|
| 0 | A   | None |
| 1 | B   | None |

pd.DataFrame.from_dict({'col': {0: 'A', 1: 'B'}, 'b': {0: None, 1: None}})

I would expect the first example to return the following DataFrame (same than example 3):

|   | col | b    |
|---|-----|------|
| 0 | A   | 100. |
| 1 | B   | None |

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 2e218d10984e9919f0296931d92ea851c6a6faf5
python           : 3.10.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.1.0
Version          : Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:30 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8103
machine          : arm64
processor        : arm
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.5.3
numpy            : 1.24.2
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 63.2.0
pip              : 23.0.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by Charlie-XIAO**
- [x] Closes #52331
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests)
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit)
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file

### CODE REVIEW

**Comment by topper-123** [comment]
Sounds great!

**Review by topper-123 — APPROVED** [review]
LGTM

File: pandas/tests/reshape/merge/test_merge.py [hunk 1]
```diff
@@ -2773,3 +2773,17 @@ def test_merge_arrow_and_numpy_dtypes(dtype):
     result = df2.merge(df)
     expected = df2.copy()
     tm.assert_frame_equal(result, expected)
+
+
+def test_merge_multiindex_single_level():
+    # Non-regression test for GH #52331
```
Comment (phofl) [inline pandas/tests/reshape/merge/test_merge.py:2779]: Can remove the comments, gh ref is sufficient

**Comment by mroeschke** [comment]
Could you resolve the merge conflict?

**Comment by Charlie-XIAO** [comment]
Done.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**